custom/icds_reports/templatetags/icds_format.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import absolute_import
+from django import template
+
+register = template.Library()
+
+
+@register.filter(name='icds_format')
+def icds_format(data, prop):
+    if data:
+        return data[prop] if data[prop] is not None else 'Data Not Entered'
+    else:
+        return ''
+
+
+@register.filter(name='icds_type_of_building')
+def icds_type_of_building(data):
+    housed = ['', 'Owned', 'Rented']
+    provided_buildings = [
+        '', 'Panchayat', 'Community', 'Urban Municipality/Corporation', ' Rural Development/ DRDA',
+        'ICDS', 'Other'
+    ]
+    if data:
+        where_housed = data['where_housed'] or 0
+        building = data['provided_building'] or 0
+        if where_housed == 3:
+            return provided_buildings[building]
+        else:
+            return housed[where_housed]
+    else:
+        return ''
+
+
+@register.filter(name='icds_yes_no')
+def icds_yes_no(value):
```

Comment:
any reason you can't just use django's [`yesno`](https://docs.djangoproject.com/en/2.0/ref/templates/builtins/#yesno)? it takes arguments if you need to customize the text.

Reply:
I didn't know that django have already this functionality :)